rest-service/manager_rest/rest/resources_v3_1/manager.py

```diff
@@ -223,12 +232,68 @@ def _get_ca_cert_id(self, sm, _manager):
             return existing_managers_certs.pop()
 
 
-class RabbitMQBrokers(SecuredResource):
+class RabbitMQBrokers(ManagementResource):
     @exceptions_handled
     @marshal_with(models.RabbitMQBroker)
     @paginate
     @authorize('broker_get')
     def get(self, pagination=None):
+        """List brokers from the database."""
         return get_storage_manager().list(
             models.RabbitMQBroker,
         )
+
+    @exceptions_handled
+    @authorize('broker_add')
+    @marshal_with(models.RabbitMQBroker)
+    def post(self):
+        """Add a broker to the database."""
+        _broker = rest_utils.get_json_and_verify_params({
+            'name': {'type': unicode},
+            'address': {'type': unicode},
+            'port': {'type': int, 'optional': True},
+            'networks': {'type': dict, 'optional': True},
+        })
+        sm = get_storage_manager()
+
+        # Get the first broker in the list to get the ca_cert and credentials
+        first_broker = sm.list(models.RabbitMQBroker, None, None).items[0]
+
+        if not _broker.get('networks'):
+            _broker['networks'] = {'default': _broker['address']}
+
+        new_broker = models.RabbitMQBroker(
+            name=_broker['name'],
+            host=_broker['address'],
+            management_host=_broker['address'],
+            port=_broker.get('port'),
+            networks=_broker['networks'],
+            username=first_broker.username,
+            password=first_broker.password,
+            ca_cert=first_broker.ca_cert,
```

Comment:
we might as well add the same ca-cert handling (with `get_ca_cert_id` etc) as we have in the managers?
What is the part that breaks if rabbitmqs have different cacerts? Agents will survive it.
Creation of the rabbitmq cluster itself breaks?

Reply:
I think the rabbitmq clustering will be broken, yes. But agreed, I'll switch over to the method used there- for some reason the manager I originally wrote this on had slightly outdated code (maybe from yesterday)

Reply:
After further discussion we've agreed that since we're not going to really support (or test) this use-case, it won't be added.